postgres/client.js

```diff
@@ -60,10 +97,38 @@ function get(key) {
  * @return {Promise}
  */
 function put(key, value) {
-  let obj = parseOrNot(value);
+  const { schema, table } = findSchemaAndTable(key);
 
-  return client.query(PUT(key), [key, wrapInObject(key, obj)])
-    .then(() => obj);
+  return onConflictPut(key, parseOrNot(value), 'data', schema, table);
+}
+
+/**
+ * Given a key (id), data, which column that
+ * data lives at, a schema and a table a `NO
+ * CONFLICT` insert is executed
+ *
+ * @param {String} id
+ * @param {Object|String} data
+ * @param {String} dataProp
+ * @param {String} schema
+ * @param {String} table
+ * @returns {Promise}
+ */
+function onConflictPut(id, data, dataProp, schema, table) {
```

Comment:
This routine is a bit outside of normal knex usage so it might be nice to annotate the constructed query in the doc comment (specifically the part where you use the query builder to generate an update statement for use within the ON CONFLICT)

Reply:
Agreed, will handle during final docs pass